src/pages/schemas/addConnectionModalBox.tsx

```diff
@@ -0,0 +1,112 @@
+import Buttons from '@/components/ComposePage/buttons/buttons';
+import { dBConnectionPostAction } from '@/redux/actions/dbConnectionAction';
+import { Input, Modal } from 'antd'
+import axios from 'axios';
+import React, { useState } from 'react'
+import { useDispatch } from 'react-redux';
+import { toast } from 'react-toastify';
+import styles from './connectionbox.module.css'
+import { showErrorToast, showSuccessToast } from './toast';
+
+
+interface MyModalProps {
+    showConnectionBox?: boolean;
+    onCancel?: () => void;
+}
+
+function showWarningToast(message: string) {
+    toast.warn(message);
+}
+
+const addConnectionModalBox: React.FC<MyModalProps> = ({ showConnectionBox, onCancel }) => {
+    const [name, setName] = useState('');
+    const [description, setDescription] = useState('');
+    const [connectionURL, setconnectionURL] = useState('');
+    const [username, setUsername] = useState('');
+    const [password, setPassword] = useState('');
+
+    const dispatch = useDispatch();
+
+    const handleOnClickCancel = () => {
+        setName('')
+        setDescription('')
+        setconnectionURL('')
+        setUsername('')
+        setPassword('')
+        onCancel()
+    }
+
+    const handleOnClickConnect = async () => {
+
+        if (!name || !connectionURL || !username || !password) {
+            showWarningToast('Please fill in all required fields.');
+            return;
+        }
+
+        const action = dispatch(
+            dBConnectionPostAction(name, description, connectionURL, username, password)
+        )
+
+        if (action) {
+            setName('')
+            setDescription('')
+            setconnectionURL('')
+            setUsername('')
+            setPassword('')
+
+            onCancel()
+            showSuccessToast("Connection Successfull")
```

Comment:
Try to keep all the hardcoded values inside a constant file. So the values can be reused.